pkg/client/cache/watch_cache.go

```diff
@@ -27,12 +27,22 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+// WatchCacheEvent is a single "watch event" that is send to users of
+// WatchCache. Additionally to a typical "watch.Event" it contains
+// the previous value of the object to enable proper filtering in the
+// upper layers.
```

Comment:
But maybe the whole "WatchCache" class shouldn't be under pkg/client/cache, but should be a private class under pkg/storage (since this is in fact an implementation detail).